src/functions-and-arrays.js

```diff
@@ -1,119 +1,261 @@
 // Iteration #1: Find the maximum
-function maxOfTwoNumbers() {}
-
-
+function maxOfTwoNumbers(number1, number2) {
+  if (number1 > number2) {
+    return number1;
+  }
+  if (number2 > number1) {
+    return number2;
+  }
+  if (number1 === number2) {
+    return number1;
+  }
+}
 
 // Iteration #2: Find longest word
-const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
-
-function findLongestWord() {}
-
+const words = [
+  "mystery",
+  "brother",
+  "aviator",
+  "crocodile",
+  "pearl",
+  "orchard",
+  "crackpot",
+];
 
+function findLongestWord(arrayData) {
+  if (arrayData.length > 0) {
+    let i = 0;
+    let biggestWord = "";
+    arrayData.forEach((el) =>
+      el.length > i ? (i = el.length) && (biggestWord = el) : ""
+    );
+    return biggestWord;
+  } else {
+    return null;
+  }
+}
 
 // Iteration #3: Calculate the sum
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
-function sumNumbers() {}
-
-
+function sumNumbers(numbers) {
+  let acc = 0;
+  numbers.forEach((el) => (acc += el));
+  return acc;
+}
 
 // Iteration #3.1 Bonus:
-function sum() {}
-
-
+function sum(data) {
+  let acc = 0;
+  if (data) {
+    data.forEach((el) => {
+      let value;
+      if (typeof el === "boolean") {
+        if (el === true) {
+          value = 1;
+        }
+        if (el === false) {
+          value = 0;
+        }
+      }
+      if (typeof el === "string") {
+        value = el.length;
+      }
+      if (typeof el === "number") {
+        value = el;
+      }
+      if (typeof el === "object" || typeof el === "array") {
+        throw "msg";
+      }
+      acc += value;
+    });
+    return acc;
+  }
+}
+sum([6, 12, "miami", 1, true, "barca", "200", "lisboa", 8, 10]);
 
 // Iteration #4: Calculate the average
 // Level 1: Array of numbers
 const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
 
-function averageNumbers() {}
-
+function averageNumbers(arrayToManipulate) {
+  if (arrayToManipulate.length < 1) return null;
+  let i = 0;
+  arrayToManipulate.forEach((el) => (i += el));
+  return i / arrayToManipulate.length;
+}
 
 // Level 2: Array of strings
-const wordsArr = ['seat', 'correspond', 'linen', 'motif', 'hole', 'smell', 'smart', 'chaos', 'fuel', 'palace'];
+const wordsArr = [
+  "seat",
+  "correspond",
+  "linen",
+  "motif",
+  "hole",
+  "smell",
+  "smart",
+  "chaos",
+  "fuel",
+  "palace",
+];
 
-function averageWordLength() { }
+function averageWordLength(arrayData) {
+  if (arrayData.length < 1) return null;
+
+  let i = 0;
+  arrayData.forEach((el) => (i += el.length));
+  return i / arrayData.length;
+}
 
 // Bonus - Iteration #4.1
-function avg() {}
+function avg(arrayData) {
```

Comment:
This iteration is similar to the 3.1 you will receive a `mixed elements array` so you need to make `conditions` checking the `typeOf` the elements:

```
function avg(array) {
  if (array.length === 0){
    return null;
  }
  let sum = 0;
  for (let i=0; i<array.length; i++){
    if (typeof array[i] === 'number'){
      sum += array[i];
    } else if (typeof array[i] === 'string'){
      sum += array[i].length;
    }else if (array[i] === true){
      sum += 1;
    }
  }
  return sum/array.length
}
```